diff -u -- q3/train.py q1/train.py
--- q3/train.py	2019-05-24 12:29:26.496210000 +0800
+++ q1/train.py	2019-05-24 11:49:05.922726300 +0800
@@ -11,9 +11,7 @@
 
 '''use 13 convolution layers to generate gram matrix'''
 
-GRAM_LAYERS= ['conv1_1', 'conv2_1', 'conv3_1', 'conv4_1' , 'conv5_1']
-# GRAM_LAYERS= ['conv1_1', 'conv1_2' ,'conv2_1', 'conv2_2']
-# GRAM_LAYERS= ['conv4_1', 'conv4_2', 'conv4_3' , 'conv5_1', 'conv5_2', 'conv5_3']
+GRAM_LAYERS= ['conv1_1', 'conv1_2' ,'conv2_1', 'conv2_2' , 'conv3_1', 'conv3_2', 'conv3_3' , 'conv4_1', 'conv4_2', 'conv4_3' , 'conv5_1', 'conv5_2', 'conv5_3']
 image_shape = (1, 224, 224, 3)
 
 '''you need to complete this method'''
@@ -58,7 +56,7 @@
     noise = tf.Variable(tf.nn.sigmoid(pre_noise))
 
     '''load texture image, notice that the pixel value has to be normalized to [0,1]'''
-    image = cv2.imread('../images/stone_1.jpg')
+    image = cv2.imread('../images/red-peppers256.jpg')
     image = cv2.resize(image, image_shape[1:3])
     image = image.reshape(image_shape)
     image = (image/255).astype('float32')
